src/main/java/uk/gov/hmcts/divorce/divorcecase/util/AddressUtil.java

```diff
@@ -57,4 +62,25 @@ public static String getPostalAddress(final AddressGlobalUK address) {
 
         return null;
     }
+
+    public static boolean isEnglandOrWales(AddressGlobalUK address) {
+        if (isNull(address) || isNull(address.getCountry())) {
+            throw new IllegalArgumentException("Cannot assert whether address is overseas or not due to null address or country");
+        }
+
+        final List<String> ukTerms = Arrays.asList("unitedkingdom", "uk", "england", "wales");
+        final List<String> scottishPostcodePrefixes =
+            Arrays.asList("ab", "dd", "dg", "eh", "fk", "g", "hs", "iv", "ka", "kw", "ky", "ml", "pa", "ph", "td", "ze");
+
+        final String sanitisedCountry = address.getCountry().replaceAll("[^a-zA-Z0-9]+", "").toLowerCase(Locale.ROOT);
+        final String postcode = Optional.ofNullable(address.getPostCode()).orElse("");
+
+        var isScottishPostcode = false;
+        if (postcode.matches(".*[a-zA-Z]+.*")) {
```

Comment:
This checks that the postcode has at least one letter in. Needed because if the postcode was numbers only the split() method below would fail.